File: src/Lucifer/Lucifer.Pms.Model/Entities/Currency.cs
using Lucifer.DataAccess;

namespace Lucifer.Pms.Model.Entities
{
    public class Currency : DomainEntity
    {
        public Currency()
        {
            DecimalChar = ',';
            ThousandChar = '.';
        }
        public virtual string Name { get; set; }
        public virtual string Contraction { get; set; }
        public virtual string Symbol { get; set; }
        public virtual decimal Rate { get; set; }
        public virtual int DecimalPosition { get; set; }
        public virtual char DecimalChar { get; set; }
        public virtual char ThousandChar { get; set; }
    }
}

File: sketches/java/JLucifer/src/main/java/de/slesa/lucifer/pms/model/Currency.java
package de.slesa.lucifer.pms.model;

import javax.persistence.Entity;
import javax.persistence.GeneratedValue;
import javax.persistence.Id;

import lombok.Data;

@Data
@Entity
public class Currency {

	@Id
	@GeneratedValue
	protected Long id;
	protected String name;
	protected String contraction;
	protected String symbol;
	protected double rate;
	protected int decimalPosition;
	protected char decimalChar;
	protected char thousandChar;
	
	public Currency(String name, String contraction) {
		this.name = name;
		this.contraction = contraction;
	}
}
